1-Draft/RFCNNNN-AdditionalTelemetry.md

```diff
@@ -0,0 +1,175 @@
+  - Unique user trends
+  - Unique devices trends
+  - Types of user &mdash;based on types of executions taking place
+  - VSCode usage trends
+  - Application trends
+  - Hosted Scenario Trends
+  - PowerShellGet installation trends
+- A cmdlet for providing feedback (Send-PSFeedback):
+  - This cmdlet would take in a string and send it back to the PowerShell team
+  - The cmdlet will also send Platform data (OS/version) and Shell data ($PSVersionTable)
+  - The cmdlet will have an -OpenGitHubIssue switch that will allow the user to open a GitHubIssue
+    in the PowerShell/PowerShell repository
+  - The user will have the option to include the the last error using an -IncludeError switch
+  - If the command is not successful (ex. there is no Web Connection) the cmdlet will return a
+    warning and alternative instructions for support through GitHub
+  - If no location switch is used the cmdlet will default send the feedback to the PowerShell
+    User Voice
+  - Based on usage of this cmdlet there would potential for additional parameters
+- Disabling, and re-enabling telemetry will be available through a new cmdlet (Set-PSTelemetry):
+  - The cmdlet will stop all collection of telemetry for the remainder of that session and stop
+    telemetry from being collected on subsequent starts
+  - The user can request their telemetry is deleted by using a Remove-PSTelemetry cmdlet
+  - In order to have their telemetry deleted the user will be required to provide their System GUID
+    and User GUID
+  - If the cmdlet is unable to send the information the cmdlet will return a warning and alternative
+    instructions for support through GitHub. It will be specified that GUIDs are unique identifiers
+    and should not be shared publicly on GitHub.
+- There will be three options for the telemetry environment variable which can all be set manually
+  or using the Set-PSTelemetry cmdlet
+  - $True indicates all telemetry will be collected
+  - $False indicates no telemetry will be collected
+  - $NonIdentifying indicates that only non-identifying telemetry will be collected i.e. no GUIDs
+    will be collected
+- The existing disabling mechanisms for PowerShell telemetry will remain including the ability to
+  set the telemetry environment variable before the first start of PowerShell
+
+### Future work
+
+A public API for module authors and PowerShell hosted application owners to use the same
+infrastructure to collect telemetry for their module or application.
+Making the API public is a phase two goal of this project.
+We will make the telemetry publicly available once it is internally validated.
+Microsoft will not collect, store, or manage any data externally gathered through this API.
+
+## Design
+
+### PowerShell Changes
+
+- Count of unique devices will be based on a System GUID generated on the first start-up
+- Count of unique users will be based on a User GUID generated at PowerShell start time and stored

```

Comment:
This needs to be expanded upon some more, I think, where these will be stored on Windows or Unix platforms, etc. I'm tossing up whether a simple `Get-PSTelemetryGuid` cmdlet should be provided -- do you think it's worth doing?

Reply:
We'll likely need some sort of `Reset-PSTelemetryGuid` as well.

Reply:
I worry about creating too many cmdlets when the GUID should be something that should only be touched by customers at most one time (if they chose to have their telemetry deleted)